src/api/v1/Auth/controller.js

```diff
@@ -1,22 +1,137 @@
 const {
     sendSuccessResp,
     sendFailureResp,
-} = require('../../../utils/response.js');
+} = require("../../../utils/response.js");
+
+const { User, Admin } = require("../../../database/index.js");
+
+const { createToken } = require("../../../services/jwt.js");
+
+const { hash, compareHash } = require("../../../services/encryption.js");
+
+const roleArray = [User, Admin];
 
 module.exports = {
-    signup: (req, res) => {
-        sendSuccessResp(res,{
-            data: {
-                message: "Mann Lo Ki User Sign UP ho gya"
+    signup: async (req, res) => {
+        const { role, name, mobile, email, password } = req.body;
+        const model = roleArray[role];
+        try {
+            const data = await findUser(role, email);
+            const hashedPassword = await hash(password);
+
+            if (!data) {
```

Comment:
use error first approch return failure response first if user already exists if not the case then write main logic don't use if else for this.

Reply:
if there is a try catch block try to throw any error is occurs , handle it only at one place inside catch block